### DISCUSSION TITLE
upgrading to v8

### DISCUSSION THREAD

**Discussion by herzaso** [storybookjs/storybook#28314] [Help]
### Summary

I've upgraded from version 7.6.10 to 8.1.10 and most things are working except one story in which I get the following error: `We detected that you use an implicit action arg while rendering of your story.`
(I did keep the `parameters: { actions: { argTypesRegex: '^on[A-Z].*' }, }` setting in the `preview` config)

I understood that there was a breaking change around actions, but several things puzzle me:
1. I have around 3 dozens of components with actions, and all of them are working well (with `actions`) except for this one component
2. I don't use testing through Storybook, I only need those actions for the `actions` tab

Do I have to go through all of my components and explicitly set all the actions (and import from `@storybook/test` although I don't use testing)?

### Additional information

**EDIT:** I pin pointed the problem to be with an action called `onUnmount`

**Comment by Bruce-Jay** [comment]
I also met with this problem and didn't know how to solve it.

**Comment by Bruce-Jay** [comment]
Oh I worked it out, just as the guidance provided by storybook official: https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#implicit-actions-can-not-be-used-during-rendering-for-example-in-the-play-function, you just have to add explicit add functions you used in the args field, and it will work.

Just like this:
![image](https://github.com/user-attachments/assets/8113a45c-be5d-4060-942d-b237fa9ef6a5)

**Reply by herzaso** [reply]
@Bruce-Jay I've read that part, and I opened it as a discussion and not as a bug since I understand the change.
My issue was that I have more than 30 stories that work with the old setting: `parameters: { actions: { argTypesRegex: '^on[A-Z].*' }, }` and only one story that fails with a prop I called `onUnmount` - this one requires the new syntax...

**Reply by Bruce-Jay** [reply]
OK, I had not met with this problem before. Maybe it's because of an another cause but not the prop named `onUnmount`? I think it's weird of failing with calling onUnmount, which is correct according to the regex.

**Reply by herzaso** [reply]
Update - it's probably not because of the name since I encountered the problem with other props I recently added (onFocus, onBlur, onSort). Still don't get the rule though (why at times it works and at times it doesn't without the use of `fn()`)

**Reply by valentinpalkovic** [reply]
cc @kasperpeulen
